CONTRIBUTING.md

```diff
@@ -1,13 +1,6 @@
 # Kubeflow Contributor Guide
```

Comment:
@thesuperzapper Would it be easier to remove this file and just add the `## Contributing` section to the README where we link the https://www.kubeflow.org/docs/about/contributing/ page ?

Reply:
We should probably have both, because google loves to index files like `CONTRIBUTING.md`.

Reply:
I added `& Contributing` to the final README heading in https://github.com/kubeflow/kubeflow/pull/7642/commits/5513f05f5a67250aeb152b4154f1bb1ed523b727